### PR TITLE
feat: add support for natspec map

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

Adds  `natspecs` mapping `ContractType`

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ethpm_types/contract_type.py [hunk 1]
```diff
@@ -469,6 +478,37 @@ def structs(self) -> ABIList:
...
+        return _extract_natspec(self.devdoc or {}, "methods", self.methods)
+
+    @cached_property
+    def _event_natspecs(self) -> Dict[str, str]:
+        # NOTE: I have only seen these appear on Solidity (not Vyper).
+        return _extract_natspec(self.devdoc or {}, "events", self.events)
+
+    @cached_property
+    def _error_natspecs(self) -> Dict[str, str]:
+        # NOTE: I have only seen these appear on Solidity (not Vyper).
```
Comment (fubuloubu) [inline ethpm_types/contract_type.py:504]: note: Vyper doesn't have custom errors yet

File: ethpm_types/contract_type.py [hunk 2]
```diff
@@ -469,6 +478,37 @@ def structs(self) -> ABIList:
...
+            **self._struct_natspecs,
+        }
+
+    @cached_property
+    def _method_natspecs(self) -> Dict[str, str]:
+        return _extract_natspec(self.devdoc or {}, "methods", self.methods)
+
+    @cached_property
+    def _event_natspecs(self) -> Dict[str, str]:
+        # NOTE: I have only seen these appear on Solidity (not Vyper).
```
Comment (fubuloubu) [inline ethpm_types/contract_type.py:499]: note: Vyper doesn't allow docstrings for event defs (but that's a good idea)

File: ethpm_types/contract_type.py [hunk 3]
```diff
@@ -469,6 +478,37 @@ def structs(self) -> ABIList:
...
+        return _extract_natspec(self.devdoc or {}, "events", self.events)
+
+    @cached_property
+    def _error_natspecs(self) -> Dict[str, str]:
+        # NOTE: I have only seen these appear on Solidity (not Vyper).
+        return _extract_natspec(self.devdoc or {}, "errors", self.errors)
+
+    @cached_property
+    def _struct_natspecs(self) -> Dict[str, str]:
+        # NOTE: I have not seen any Struct NatSpecs show up in compiler outputs
```
Comment (fubuloubu) [inline ethpm_types/contract_type.py:509]: note: Vyper doesn't allow docstrings for struct defs (but that's a good idea)

File: ethpm_types/contract_type.py [hunk 4]
```diff
@@ -469,6 +478,37 @@ def structs(self) -> ABIList:
         """
         return self._get_abis(filter_fn=lambda a: isinstance(a, StructABI))
 
+    @property
```
Comment (fubuloubu) [inline ethpm_types/contract_type.py:481]: this seems heavy, should it be cached?

**Review by fubuloubu — APPROVED** [review]

File: ethpm_types/contract_type.py (hunk 2 above)
Comment (antazoey) [reply ethpm_types/contract_type.py:499]: I see snekmate does it anyway! They just don't show up in the devdoc

Comment (fubuloubu) [reply ethpm_types/contract_type.py:499]: Yes, there's an issue tracking

File: ethpm_types/contract_type.py (hunk 1 above)
Comment (antazoey) [reply ethpm_types/contract_type.py]: all comments adjusted since now i know for certain!

**Pull request merged**